src/include/stir/KOSMAPOSL/KOSMAPOSLReconstruction.h

```diff
@@ -0,0 +1,291 @@
+
+    /*! \name Functions to set parameters
+    This can be used as alternative to the parsing mechanism.
+   \warning Be careful with setting shared pointers. If you modify the objects in 
+   one place, all objects that use the shared pointer will be affected.
+  */
+
+  void set_anatomical_prior_sptr(shared_ptr<TargetT>&);
+
+
+  //! boolean value to determine if the update images have to be written to disk
+  void set_write_update_image(const int);
+
+
+  //! prompts the user to enter parameter values manually
+  virtual void ask_parameters();
+
+ protected:
+  //! Filename with input projection data
+  std::string input_filename,kernelised_output_filename_prefix;
+  std::string current_kimage_filename;
+  std::string sens_filenames;
+
+  //! Anatomical image filename
+  std::vector<std::string> anatomical_image_filenames;
+
+  shared_ptr<TargetT> anatomical_prior_sptr;
+  shared_ptr<TargetT> kpnorm_sptr,kmnorm_sptr;
+ //kernel parameters
+  int num_neighbours,num_non_zero_feat,num_elem_neighbourhood,num_voxels,dimz,dimy,dimx;
+  double sigma_m;
+  bool only_2D;
+  bool hybrid;
+  double sigma_p;
+  double sigma_dp, sigma_dm;
+
+  //! boolean value to determine if the update images have to be written to disk
+  int write_update_image;
+
+    virtual void set_defaults();
+  virtual void initialise_keymap();
+
+  //! used to check acceptable parameter ranges, etc...
+  virtual bool post_processing();
+
+ 
+private:
+  friend void do_sensitivity(const char * const par_filename);
+
+  //! operations prior to the iterations
+  virtual Succeeded set_up(shared_ptr <TargetT > const& target_image_ptr);
+ 
+  //! the principal operations for updating the image iterates at each iteration
+  virtual void update_estimate (TargetT& current_image_estimate);
+
+  int subiteration_counter;
+  double anatomical_sd;
+  mutable Array<3,float> distance;
+  /*! Create a matrix containing the norm of the difference between two feature vectors, \f$ \|  \boldsymbol{z}^{(n)}_j-\boldsymbol{z}^{(n)}_l \| \f$. */
+  /*! This is done for the emission image which keeps changing*/
+    void  calculate_norm_matrix(TargetT &normp,
+                                const int dimf_row,
+                                const int dimf_col,
+                                const TargetT& emission);
+
+  /*! Create a matrix similarly to calculate_norm_matrix() but this is done for the anatomical image, */
+  /*! which does not  change over iteration.*/
+    void calculate_norm_const_matrix(TargetT &normm,
+                                const int dimf_row,
+                                const int dimf_col);
+
+  /*! Estimate the SD of the anatomical image to be used as normalisation for the feature vector */
+    double estimate_stand_dev_for_anatomical_image();
+
+  /*! Compute for each voxel, jl, of the emission image the linear combination between the coefficient \f$ \alpha_{jl} \f$ and the kernel matrix \f$ k_{jl} \f$\f$ */
```

Comment:
get rid of these lines?
in any case, the remaining doxygen lines will likely mess up the doxygen doc